### ISSUE TITLE
Document and maybe consolidate duration parameter format

### ISSUE THREAD

**Issue by grobie** [prometheus/prometheus#1883]
The prometheus server accepts a couple of time duration parameters. These currently only allow the golang time.Duration units, with `h` being the largest one. On the other side, promql also accepts d/w/y etc.

So far, the valid format is not document anywhere, neither in the usage nor on prometheus.io. We should at least document that.

I'd also vote to consolidate the formats and accept the same format everywhere. There are arguments that any unit larger than hour can be ambiguous due to leap seconds/days. Personally I'd argue a leap seconds and days don't matter when someone specifies large retention intervals.

**Comment by brian-brazil** [comment]
I was of the impression that we'd already done all of that.

**Comment by beorn7** [comment]
Flags are still using vanilla Go duration.

A problem is that PromQL and config durations are not a superset of the Go durations. The former supports d, w, y. The latter can do mixed units like 5m30s.

**Comment by grobie** [comment]
Do you think we actually need mixed units in flags? I can't think of a
reasonable use case immediately, and one can always drop down to the next
lower unit.

**Comment by beorn7** [comment]
I'm fine with not allowing mixed units. But it could break existing setups, and that's why we didn't just flip the switch.

If we allowed mixed units, we would have a non-breaking upgrade path.

**Comment by brian-brazil** [comment]
I think we're okay with switching to not having mixed units for the sake of consistency, and such a breaking change is permitted within 1.0.

**Comment by grobie** [comment]
I don't think we can break the existing usage. We explicitly promised
"Exact flag meanings may change as necessary. However, changes will never
cause the server to not start with previous flag configurations.". I'd
print a warning log line in that case and convert them.

**Comment by grobie** [comment]
Yes, in case we still have time duration flags left. I didn't do it due to my concerns about compatibility.

**Comment by fabxc** [comment]
So the plan would to to switch to our single unit while supporting days, weeks, and years? Where does this need changes. So far I got:

- `storage.tsdb.retention`
- `storage.tsdb.min-block-duration`
- `storage.tsdb.max-block-duration`

Anything I missed?

**Comment by grobie** [comment]
@fabxc I'm counting 7 DurationVars.

**Comment by Gouthamve** [comment]
So do we make this change only to those flags requiring long intervals or to all `Duration` flags for consistency?

I am talking about:
```
--query.lookback-delta
--query.timeout
--web.read-timeout
--alertmanager.timeout
```

**Comment by grobie** [comment]
+1 to all.

**Comment by fabxc** [comment]
Yea, that came to my mind as well. But all seems saner and I cannot really see a use for mixed units.

**Comment by Gouthamve** [comment]
@brian-brazil @grobie Is only documenting it left as we moved all the flags? Or do we need to change anything in promql?

**Comment by grobie** [comment]
It's not about promql, it's about flags. I don't know what the current state is in 2.0, but in 1.x we have for example `-storage.local.retention 360h0m0s`. This is the standard golang time format. The request was to consolidate all of these and use our PromQL duration format, allowing to set `-storage.local.retention 15d`.

**Comment by Gouthamve** [comment]
Then I think this can be closed as it is now in 2.0:
```
➜  prometheus git:(dev-2.0) ✗ ./prometheus --storage.tsdb.retention=15d
INFO[0000] Starting prometheus (version=, branch=, revision=)  source="main.go:202"
INFO[0000] Build context (go=go1.8.3, user=, date=)      source="main.go:203"
INFO[0000] Host details (darwin)                         source="main.go:204"
INFO[0000] Starting tsdb                                 source="main.go:216"
```

**Comment by grobie** [comment]
Cool, the acceptance criteria is basically: all duration flags should use
our PromQL style duration format and not the golang style one. Feel free to
close if that's the case! Thanks for checking.

**Comment by Gouthamve** [comment]
Fixed in #2844
